kobuki_node/include/kobuki_node/kobuki_ros.hpp

```diff
@@ -44,111 +44,134 @@
  ** Includes
  *****************************************************************************/
 
+#include <memory>
 #include <string>
-#include <boost/shared_ptr.hpp>
-
-#include <ros/ros.h>
-#include <angles/angles.h>
-#include <std_msgs/Empty.h>
-#include <std_msgs/String.h>
-#include <std_msgs/Int16MultiArray.h>
-#include <sensor_msgs/JointState.h>
-#include <sensor_msgs/Imu.h>
+#include <vector>
+
+#include <diagnostic_updater/diagnostic_updater.hpp>
+#include <nav_msgs/msg/odometry.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/imu.hpp>
+#include <sensor_msgs/msg/joint_state.hpp>
+#include <std_msgs/msg/empty.hpp>
+#include <std_msgs/msg/int16_multi_array.hpp>
+#include <std_msgs/msg/string.hpp>
+#include <tf2_ros/transform_broadcaster.h>
+
 #include <ecl/sigslots.hpp>
-#include <kobuki_msgs/ButtonEvent.h>
-#include <kobuki_msgs/BumperEvent.h>
-#include <kobuki_msgs/CliffEvent.h>
-#include <kobuki_msgs/ControllerInfo.h>
-#include <kobuki_msgs/DigitalOutput.h>
-#include <kobuki_msgs/DigitalInputEvent.h>
-#include <kobuki_msgs/ExternalPower.h>
-#include <kobuki_msgs/DockInfraRed.h>
-#include <kobuki_msgs/Led.h>
-#include <kobuki_msgs/MotorPower.h>
-#include <kobuki_msgs/PowerSystemEvent.h>
-#include <kobuki_msgs/RobotStateEvent.h>
-#include <kobuki_msgs/SensorState.h>
-#include <kobuki_msgs/Sound.h>
-#include <kobuki_msgs/VersionInfo.h>
-#include <kobuki_msgs/WheelDropEvent.h>
+
+#include <kobuki_ros_interfaces/msg/button_event.hpp>
+#include <kobuki_ros_interfaces/msg/bumper_event.hpp>
+#include <kobuki_ros_interfaces/msg/cliff_event.hpp>
+#include <kobuki_ros_interfaces/msg/controller_info.hpp>
+#include <kobuki_ros_interfaces/msg/digital_output.hpp>
+#include <kobuki_ros_interfaces/msg/digital_input_event.hpp>
+#include <kobuki_ros_interfaces/msg/external_power.hpp>
+#include <kobuki_ros_interfaces/msg/dock_infra_red.hpp>
+#include <kobuki_ros_interfaces/msg/led.hpp>
+#include <kobuki_ros_interfaces/msg/motor_power.hpp>
+#include <kobuki_ros_interfaces/msg/power_system_event.hpp>
+#include <kobuki_ros_interfaces/msg/robot_state_event.hpp>
+#include <kobuki_ros_interfaces/msg/sensor_state.hpp>
+#include <kobuki_ros_interfaces/msg/sound.hpp>
+#include <kobuki_ros_interfaces/msg/version_info.hpp>
+#include <kobuki_ros_interfaces/msg/wheel_drop_event.hpp>
+
 #include <kobuki_driver/kobuki.hpp>
+
 #include "diagnostics.hpp"
 #include "odometry.hpp"
 
 /*****************************************************************************
  ** Namespaces
  *****************************************************************************/
 
-namespace kobuki
+namespace kobuki_node
 {
-class KobukiRos
+class KobukiRos final : public rclcpp::Node
```

Comment:
Either way is fine by me to move it forward, but I'm curious - are there any constraints to the _has-a_ (composition) vs _is-a_ (inheritence) style in rclcpp?

Reply:
It's a good question.  My understanding was that "is-a" is strictly better than "has-a" in ROS 2 land, since "is-a" can be composed.  But I'll ask around.

Reply:
Ah, I'd forgotten that - that makes this change quite necessary then.